target/linux/ath79/dts/ar9344_netgear_pgzng1.dts

```diff
@@ -0,0 +1,341 @@
+				color = <LED_COLOR_ID_RED>;
+				function = LED_FUNCTION_INDICATOR;
+				type = <PCA955X_TYPE_LED>;
+			};
+		};
+
+		expander1: pca9551@61 { 
+			compatible = "nxp,pca9551";
+			reg = <0x61>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			gpio-controller;
+			#gpio-cells = <2>;
+
+			/* zwave_rst - Resets ZWave */
+			gpio@4 {
+				reg = <4>;
+				type = <PCA955X_TYPE_GPIO>;
+			};
+
+			/* em_rst - Unknown */
+			gpio@5 {
+				reg = <5>;
+				type = <PCA955X_TYPE_GPIO>;
+			};
+
+			/* tp34 - Test point on PCB? */
+			gpio@6 {
+				reg = <6>;
+				type = <PCA955X_TYPE_GPIO>;
+			};
+
+			/* sw_rst - resets SoC */
+			gpio@7 {
+				reg = <7>;
+				type = <PCA955X_TYPE_GPIO>;
+			};
+		};
+
+		rtc@6f {
+			compatible = "isil,isl1208";
+			reg = <0x6f>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		eth1_link {
+			label = "green:eth1";
+			gpios = <&gpio 0 GPIO_ACTIVE_HIGH>;
+		};
+
+		eth1_act {
+			label = "amber:eth1";
+			gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+		};
+
+		eth0_link {
+			label = "green:eth0";
+			gpios = <&gpio 2 GPIO_ACTIVE_HIGH>;
+		};
+
+		eth0_act {
+			label = "amber:eth0";
+			gpios = <&gpio 3 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&eth0 {
+    status = "okay";
+	phy-handle = <&swphy0>;
+
+	nvmem-cells = <&macaddr_caldata_0>;
+	nvmem-cell-names = "mac-address";
+};
+
+&eth1 {
+    status = "okay";
+    phy-handle = <&swphy4>;
+
+	nvmem-cells = <&macaddr_caldata_6>;
+	nvmem-cell-names = "mac-address";
+
+    gmac-config {
+        device = <&gmac>;
+        switch-phy-swap = <1>;
+        switch-only-mode = <1>;
+    };
+};
+
+&gpio {
+	gpio_ext_lna0 {
+		gpio-hog;
+		line-name = "ext:lna0";
+		gpios = <18 GPIO_ACTIVE_HIGH>;
+		output-high;
+	};
+
+	gpio_ext_lna1 {
+		gpio-hog;
+		line-name = "ext:lna1";
+		gpios = <19 GPIO_ACTIVE_HIGH>;
+		output-high;
+	};
+
+	gpio_enable_rs422 {
+		gpio-hog;
+		line-name = "power:rs422";
+		gpios = <20 GPIO_ACTIVE_HIGH>;
+		output-high;
+	};
+};
+
+&nand {
+	status = "okay";
+
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		partition@0 {
+			label = "u-boot";
+			reg = <0x0 0x40000>;
+			read-only;
+		};
+
+		partition@40000 {
+			label = "u-boot-env";
+			reg = <0x40000 0x40000>;
+			read-only;
+		};
+
+		/* factory kernel used to be reg = <0x80000 0x200000> */
+		partition@80000 {
+			label = "kernel";
+			reg = <0x80000 0x500000>;
+		};
+
+		/* factory rootfs used to be reg = <0x280000 0x5000000> */
+		partition@580000 {
+			label = "ubi";
+			reg = <0x580000 0x4d00000>;
+		};
+
+		partition@5280000 {
+			label = "uImage2";
+			reg = <0x5280000 0x200000>;
+			read-only;
+		};
+
+		partition@5480000 {
+			label = "rootfs_bak";
+			reg = <0x5480000 0x5000000>;
+			read-only;
+		};
+
+		partition@a480000 {
+			label = "config";
+			reg = <0xa480000 0x1400000>;
+			read-only;
+		};
+
+		partition@b880000 {
+			label = "storage";
+			reg = <0xb880000 0x4700000>;
+			read-only;
+		};
+
+		partition@ff80000 {
```

Comment:
Hmm

some (really basic) NAND chips expose their bad block table at the end. Could this be padding the bbt (instead of a "dummy partition"?) You can check by looking at the content. If it's all FF then it's likely not used.

Reply:
It's all FF's, but I don't believe it's a bad block table. I looked at the Datasheet for the NAND (SK Hynix h27u2g8f2ctr) and don't see a callout mentioning this.

I am going to leave this in for now, but I believe it is indeed leftover from whatever SDK was used by Netgear in case older NAND packages were used.